cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/visitor/AbstractKnowledgeArtifactVisitor.java

```
@@ -32,4 +38,27 @@ protected List<IBaseBackboneElement> findArtifactCommentsToUpdate(
             throw new UnprocessableEntityException("Version not supported");
         }
     }
+
+    protected List<IDomainResource> getComponents(
+            KnowledgeArtifactAdapter adapter, Repository repository, ArrayList<IDomainResource> resourcesToUpdate) {
+        adapter.getRelatedArtifactsOfType("composed-of").stream().forEach(c -> {
+            final var preReleaseReference = KnowledgeArtifactAdapter.getRelatedArtifactReference(c);
+            Optional<KnowledgeArtifactAdapter> maybeArtifact =
+                    VisitorHelper.tryGetLatestVersion(preReleaseReference, repository);
+            if (maybeArtifact.isPresent()) {
```

these artifacts are owned so if they're not present maybe we throw an error?